tests/cbmc/proofs/s2n_free/s2n_free_harness.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "utils/s2n_blob.h"
+#include "error/s2n_errno.h"
+
+#include <assert.h>
+#include <cbmc_proof/proof_allocators.h>
+#include <cbmc_proof/make_common_datastructures.h>
+#include <cbmc_proof/cbmc_utils.h>
+
+int munlock(const void *addr, size_t len)
+{
+    assert(S2N_MEM_IS_WRITABLE(addr,len));
+    return nondet_int();
+}
+
+int s2n_calculate_stacktrace() { return nondet_int(); }
+
+long sysconf(int name) { return nondet_long(); }
+
+void s2n_free_harness() {
+    struct s2n_blob *blob = cbmc_allocate_s2n_blob();
+
+    __CPROVER_assume(s2n_blob_is_valid( blob ));
+
+    /* Non-deterministically set initialized (in s2n_mem) to true. */
+    if(nondet_bool()) {
+        s2n_mem_init();
+    }
+
+    if (s2n_free(blob) == 0) {
```

Comment:
IF this fails, what are the semantics? That the blob is unchanged? Or is the blob simply any unknown value?

Reply:
Updated the s2n_free function and proof to verify this behavior. Regardless of whether there is a failure, the memory will be wiped. The proof verifies this.

Reply:
nit: `s/0/S2N_SUCCESS/g`